### PR TITLE
Wait for working internet access before connecting

### PR DESCRIPTION
**Pull request by Rafficer**
This PR contains 2 prominent changes:

1. Splitting up the API endpoints and the domain
2. Add a "waiting for connection" feature to allow the program to run on boot and not fail when the internet connection isn't up and running already

It also updates the guide for auto connect with a systemd service to use this feature.

### CODE REVIEW

File: protonvpn_cli/utils.py [hunk 1]
```diff
@@ -19,8 +19,12 @@
 )
 
 
-def call_api(url, json_format=True):
-    """Call to the ProtonMail API at https://api.protonmail.ch."""
+def call_api(endpoint, json_format=True, handle_errors=True):
+    """Call to the ProtonVPN API."""
+
+    api_domain = "https://api.protonmail.ch"
```
Comment (kaplun) [inline protonvpn_cli/utils.py]: It must be api.protonvpn.ch

Comment (kaplun) [reply protonvpn_cli/utils.py]: It really would be better to make it configurable.

File: protonvpn_cli/utils.py [hunk 2]
```diff
@@ -29,21 +33,26 @@ def call_api(url, json_format=True):
...
         response = requests.get(url, headers=headers)
     except (requests.exceptions.ConnectionError,
             requests.exceptions.ConnectTimeout):
         print(
-            "[!] There was an error connecting to the ProtonMail API.\n"
+            "[!] There was an error connecting to the ProtonVPN API.\n"
             "[!] Please make sure your connection is working properly!"
         )
-        logger.debug("Error connecting to ProtonMail API")
+        logger.debug("Error connecting to ProtonVPN API")
```
Comment (kaplun) [inline protonvpn_cli/utils.py]: Why is this `logger.debug` and not `logger.error`?

File: protonvpn_cli/utils.py [hunk 3]
```diff
@@ -29,21 +33,26 @@ def call_api(url, json_format=True):
...
         sys.exit(1)
     try:
         response.raise_for_status()
     except requests.exceptions.HTTPError:
         print(
-            "[!] There was an error with accessing the ProtonMail API.\n"
+            "[!] There was an error with accessing the ProtonVPN API.\n"
```
Comment (kaplun) [inline protonvpn_cli/utils.py]: In fact now that I am realizing: all the errors should be logged as error (on `stderr`)

File: protonvpn_cli/utils.py [hunk 4]
```diff
@@ -180,6 +189,23 @@ def is_connected():
...
+    print("Waiting for connection...")
+    while True:
+        try:
+            logger.debug("Waiting for connection...")
+            call_api("/test/ping", handle_errors=False)
+            time.sleep(2)
+            print("Connection working!")
+            logger.debug("Connection working!")
+            break
+        except (requests.exceptions.ConnectionError,
```
Comment (kaplun) [inline protonvpn_cli/utils.py]: Don't we want to exit after a while?

File: protonvpn_cli/utils.py (hunk 2 above)
Comment (Rafficer) [reply protonvpn_cli/utils.py]: I've created #48 for this to keep track of it. This should be done program wide in another commit

File: protonvpn_cli/utils.py (hunk 1 above)
Comment (Rafficer) [reply protonvpn_cli/utils.py]: As discussed in #42 :)

File: protonvpn_cli/utils.py (hunk 4 above)
Comment (Rafficer) [reply protonvpn_cli/utils.py]: It automatically stops after 20 minutes now

File: protonvpn_cli/utils.py [hunk 5]
```diff
@@ -184,6 +189,30 @@ def is_connected():
     return True if ovpn_processes != [] else False
 
 
+def wait_for_network():
+    """Check if internet access is working"""
+
+    print("Waiting for connection...")
+    max_waiting_time = 20 * 60  # 20 Minutes
```
Comment (kaplun) [inline protonvpn_cli/utils.py]: This is quite a long timeout for waiting computer to boot. Wouldn't it be better to stick to e.g. 2 minutes by default? Also is one of the magic numbers that need to be made configurable (I guess in a next iteration?)

Comment (Rafficer) [reply protonvpn_cli/utils.py]: I'll make it configurable with the environment variable `PVPN_WAIT` that's needed anyway to start it. Then e.g. `PVPN_WAIT=300` makes it wait for 5 minutes

Comment (Rafficer) [reply protonvpn_cli/utils.py]: Added this as I've proposed with the latest commit

**Pull request merged**